packages/opentelemetry-propagator-jaeger/src/JaegerHttpTracePropagator.ts

```diff
@@ -55,31 +59,67 @@ export class JaegerHttpTracePropagator implements TextMapPropagator {
 
   inject(context: Context, carrier: unknown, setter: TextMapSetter) {
     const spanContext = getSpanContext(context);
-    if (!spanContext || isInstrumentationSuppressed(context)) return;
-
-    const traceFlags = `0${(spanContext.traceFlags || TraceFlags.NONE).toString(
-      16
-    )}`;
-
-    setter.set(
-      carrier,
-      this._jaegerTraceHeader,
-      `${spanContext.traceId}:${spanContext.spanId}:0:${traceFlags}`
-    );
+    const baggage = getBaggage(context);
+    if (spanContext && isInstrumentationSuppressed(context) === false) {
+      const traceFlags = `0${(
+        spanContext.traceFlags || TraceFlags.NONE
+      ).toString(16)}`;
+
+      setter.set(
+        carrier,
+        this._jaegerTraceHeader,
+        `${spanContext.traceId}:${spanContext.spanId}:0:${traceFlags}`
+      );
+    }
+
+    if (baggage) {
+      for (const [key, entry] of baggage.getAllEntries()) {
+        setter.set(
+          carrier,
+          `${UBER_BAGGAGE_HEADER_PREFIX}-${key}`,
+          encodeURIComponent(entry.value)
+        );
+      }
+    }
   }
 
   extract(context: Context, carrier: unknown, getter: TextMapGetter): Context {
     const uberTraceIdHeader = getter.get(carrier, this._jaegerTraceHeader);
     const uberTraceId = Array.isArray(uberTraceIdHeader)
       ? uberTraceIdHeader[0]
       : uberTraceIdHeader;
-
-    if (typeof uberTraceId !== 'string') return context;
-
-    const spanContext = deserializeSpanContext(uberTraceId);
-    if (!spanContext) return context;
-
-    return setSpanContext(context, spanContext);
+    const baggageValues = getter
+      .keys(carrier)
+      .filter(key => key.includes(UBER_BAGGAGE_HEADER_PREFIX))
```

Comment:
I think this should be `startsWith(`${UBER_BAGGAGE_HEADER_PREFIX}-`. Maybe even check that it is longer then ``${UBER_BAGGAGE_HEADER_PREFIX}-` otherwise it could happen that the `key.replace` below returns an empty string.

Reply:
or check `/^uberctx-(.+)/i` match